### DISCUSSION TITLE
How to directly save a PDF instead of seeing a preview - alternative: saving a PDF directly without save-as-dialog

### DISCUSSION THREAD

**Discussion by VaporiZed** [MicrosoftEdge/WebView2Feedback#4047] [Q&A]
[I already asked the VisualStudio team](https://developercommunity.visualstudio.com/t/Cant-configure-WebView2-in-desired-way/10362251), but they sent me here.

There are websites with links to PDF files.
I'd like to save a linked PDF file directly when clicking it's link instead of getting a preview of it, but I can't find the option to activate it.
There is also no just-save-the-file-in-the-default-directory-button. The save-button creates a save-as-dialog - not what I expected.
I also tried to configure the WebView2-instance via `CallDevToolsProtocolMethodAsync` and `Browser.setDownloadBehavior` (by thinking that I could achieve the desired behavior on this way) - but I couldn't find out, how to make it right. All the code trials led to "value out of range" errors.
And finally my question: How to make it right?

**Comment by vickiez** [comment]
@VaporiZed Can you try using window.open to download the PDF? See the example here: https://github.com/MicrosoftEdge/WebView2Feedback/issues/1127#issuecomment-814221009. You also may be able to accomplish this using WebResourceRequested event (thread here: https://github.com/MicrosoftEdge/WebView2Feedback/issues/38#issuecomment-713392975).

As far as bypassing the Save dialog, we're currently developing a SaveAs API in #1090 that may address your scenario. cc: @Master-Ukulele

**Comment by VaporiZed** [comment]
Yes, the javascript-command `WebViewInstance.CoreWebView2.ExecuteScriptAsync("window.open('here-the-link')")` is usable - if I have a link. Unfortunately, some websites use JS for themselves to show some PDFs, so I don't think, that I get the links. Sadly, I can only see the PDF previews in WebView2.
